### PR TITLE
core: remove kernel_types.h

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

This PR removes core/include/kernel_types.h, which seems to only contain obsolete info.
Another commit removes any includes of that file.

~~This also pointed to a missing include for inttypes.h in sched.h, which is added in another commit.~~ moved to #15481.

### Testing procedure

Successful CI should be sufficient.

### Issues/PRs references

Follow-up of #14787

### CODE REVIEW

**Comment by kaspar030** [comment]
Let's see what CI uncovers.

**Comment by kaspar030** [comment]
> Let's see what CI uncovers.

turns out it is not that simple. :/

File: pkg/ccn-lite/patches/0001-riot-adapt-to-changed-kernel_types.h-sched.h.patch [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+---
+ src/ccnl-core/include/ccnl-if.h       | 2 +-
+ src/ccnl-riot/include/ccn-lite-riot.h | 2 +-
+ src/ccnl-riot/src/ccn-lite-riot.c     | 2 +-
+ 3 files changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/src/ccnl-core/include/ccnl-if.h b/src/ccnl-core/include/ccnl-if.h
+index cf90dcf9c3..40d461a55e 100644
+--- a/src/ccnl-core/include/ccnl-if.h
++++ b/src/ccnl-core/include/ccnl-if.h
```
Comment (benpicco) [inline pkg/ccn-lite/patches/0001-riot-adapt-to-changed-kernel_types.h-sched.h.patch:15]: please also provide an upstream pr :)

File: sys/include/can/router.h [hunk 2]
```diff
@@ -24,7 +24,6 @@
 extern "C" {
 #endif
 
-#include "kernel_types.h"
```
Comment (benpicco) [inline sys/include/can/router.h:27]: nitpick: now there is one newline too much ;)

File: cpu/native/include/native_internal.h [hunk 3]
```diff
@@ -50,7 +50,6 @@
 #include <sys/stat.h>
 #include <sys/uio.h>
 
-#include "kernel_types.h"
```
Comment (benpicco) [inline cpu/native/include/native_internal.h:53]: here too

File: cpu/lpc23xx/periph/rtc.c [hunk 4]
```diff
@@ -24,7 +24,6 @@
 #include <stdint.h>
 #include <string.h>
 
-#include "kernel_types.h"
```
Comment (benpicco) [inline cpu/lpc23xx/periph/rtc.c:27]: and here

**Comment by kaspar030** [comment]
I fixed the extra newlines.

**Review by benpicco — APPROVED** [review]
looks good to me - please squash.
Let's see if it now also looks good to Murdock.

**Comment by kaspar030** [comment]
Thanks for the review!

**Pull request merged**